### PR TITLE
feat: migration - add SWM dashboard fetching, update UX

### PR DESCRIPTION
**Pull request by dpportet**
# Description

* Adding the logic for fetching the SWM dashboards and creating application dashboards from them
* Updated UX after feedback to use flashbars instead of status indicators 
* TODOs 
   * Find a better way to dedupe the flashbars for migration so only 1 is shown at a time
   * Add more testing on the migration service after adding proper error handling 
   * Testing for pagination, I was having trouble getting it to work. 

Fixes # (issue)

## Type of change

Please delete options that are not relevant.
- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Manually tested, unit tests

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

## Legal

This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: apps/core/src/migration/util/convertSiteWiseMonitorToApplicationDefinition.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { DashboardDefinition } from 'src/dashboards/entities/dashboard-definition.entity';
+
+export const convertSiteWiseMonitorToApplicationDefinition = (
+  sitewiseMonitorDashboardDefinition?: string,
+): DashboardDefinition => {
+  // TODO: add actual definition conversion
```
Comment (corteggiano) [inline apps/core/src/migration/util/convertSiteWiseMonitorToApplicationDefinition.ts:6]: assuming this will be a followup PR?

Comment (dpportet) [reply apps/core/src/migration/util/convertSiteWiseMonitorToApplicationDefinition.ts:6]: Yes

Comment (dpportet) [reply apps/core/src/migration/util/convertSiteWiseMonitorToApplicationDefinition.ts:6]: This part will be the most tricky part of the migration so will take some time, so I wanted to get in the rest first.

**Review by corteggiano — APPROVED** [review]
lgtm!

**Pull request merged**